messaging/src/main/java/org/axonframework/configuration/NewConfigurer.java

```diff
@@ -82,46 +83,50 @@ <C> S registerComponent(@Nonnull Class<C> type,
      * Registers a {@link Component} {@link ComponentDecorator decorator} that will act on
      * {@link #registerComponent(Class, ComponentFactory) registered} components of the given {@code type}.
      * <p>
-     * The {@code order} parameter dictates at what point in time the given {@code decorator} is invoked during
-     * construction of the {@code Component} it decorators. If a {@code ComponentDecorator} was already present at the
-     * given {@code order}, it will be replaced by the given {@code decorator}
+     * Decorators are invoked based on the given {@code order}. Decorators with a lower {@code order} will be executed
+     * before those with a higher one. If decorators depend on the result of another decorator, their {@code order} must
+     * be strictly higher than the one they depend on. The order in which components are decorated by decorators with
+     * the same {@code order} is undefined.
      *
      * @param type      The declared type of the component to decorate, typically an interface.
-     * @param order     The order of the given {@code decorator} among other decorators. Becomes important whenever
-     *                  multiple decorators are present for the given {@code type} <b>and</b> when ordering of these
-     *                  decorators is important.
+     * @param order     The order of the given {@code decorator} among other decorators.
      * @param decorator The decoration function of this component.
      * @param <C>       The type of component the {@code decorator} decorates.
      * @return The current instance of the {@code Configurer} for a fluent API.
      */
-    default <C> S registerDecorator(@Nonnull Class<C> type,
-                                    int order,
-                                    @Nonnull ComponentDecorator<C> decorator) {
-        return registerDecorator(type, type.getSimpleName(), order, decorator);
-    }
+    <C> S registerDecorator(@Nonnull Class<C> type,
```

Comment:
I think it would be good if the JavaDoc of this method clarifies it acts on all Components for the given `type`, regardless of their name.